### ISSUE TITLE
Some deprecations on Julia 0.7

### ISSUE THREAD

**Issue by JockLawrie** [cjprybol/uCSV.jl#9]
Hi there,

Love this package!
FYI, I ran the following code on Julia 0.7 and got the deprecation warnings below.

```julia
using DataFrames
using uCSV

#d is my DataFrame

filename = "test.tsv"
hdr = String.(names(d))
uCSV.write(filename, d; header=hdr, delim='\t')

a    = Dict(String(colname) => typeof(eltype(d[colname])) <: Union for colname in names(d))
enc  = Dict("missing" => missing)
tp   = Dict(String(colname) => eltype(d[colname]) for colname in names(d))
tprs = Dict(Char => (x) -> x[1])
d2   = DataFrame(uCSV.read(filename; delim='\t', header=1, allowmissing=a, encodings=enc, types=tp, typeparsers=tprs))
```

Here are the warnings:

```
julia> d2   = DataFrame(uCSV.read(filename; delim='\t', header=1, allowmissing=a, encodings=enc, types=tp, typeparsers=tprs))                                 
┌ Warning: `findfirst(A, v)` is deprecated, use `something(findfirst(isequal(v), A), 0)` instead.                                                             
│   caller = (::getfield(uCSV, Symbol("##56#59")){Array{String,1}})(::Pair{String,Type}) at none:0                                                            
└ @ uCSV ./none:0                                                                                                                                             
┌ Warning: `findfirst(A, v)` is deprecated, use `something(findfirst(isequal(v), A), 0)` instead.                                                             
│   caller = #56 at none:0 [inlined]                                                                                                                          
└ @ Core ./none:0                                                                                                                                             
┌ Warning: `findfirst(A, v)` is deprecated, use `something(findfirst(isequal(v), A), 0)` instead.                                                             
│   caller = iterate at none:0 [inlined]                                                                                                                      
└ @ Core ./none:0                                                                                                                                             
┌ Warning: `Array{T, 1}(m::Int) where T` is deprecated, use `Array{T, 1}(undef, m)` instead.                                                                  
│   caller = (::getfield(uCSV, Symbol("##31#47")){Int64})(::Tuple{Int64,Union}) at none:0                                                                     
└ @ uCSV ./none:0                                                                                                                                             
┌ Warning: `Array{T, 1}(m::Int) where T` is deprecated, use `Array{T, 1}(undef, m)` instead.                                                                  
│   caller = (::getfield(uCSV, Symbol("##31#47")){Int64})(::Tuple{Int64,DataType}) at none:0                                                                  
└ @ uCSV ./none:0 
```

Cheers,
Jock

**Comment by JockLawrie** [comment]
...and one more:

```julia
┌ Warning: `warn()` is deprecated, use `@warn` instead.
│   caller = #warn#781(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::String) at deprecated.jl:1032
└ @ Base ./deprecated.jl:1032
```

**Comment by cjprybol** [comment]
Hi Jock, I'm glad this package has been helpful and you enjoy using it! I noticed these warnings when I was making the Julia 0.7 updates for the current release of uCSV, but I wasn't able to track down where they were coming from. 

The `@ uCSV ./none:0` error wasn't helpful as there is no file named `none` in the uCSV code. I think that might be code that is generated during function compilation? I think the other warning is from Julia itself, if it is coming from `@ Base ./deprecated.jl:1032`, but I'm not sure. I couldn't trace these back to any specific line of code in uCSV, but if you have any clues for where they might be coming from, please let me know!

**Comment by JockLawrie** [comment]
I think I found them all using the `ack` command line utility.
```bash
$ cd /path/to/package
$ ack warn
```

The results are:
src/read.jl, line 213
src/helperfunctions.jl, line 46

`ack findfirst` finds `src/helperfunctions.jl`, line 27.

`ack Vector` finds `src/parsesource.jl`, line 178 (among a whole lot of other stuff).

`Vector{Int64}()` returns a vector with 0 elements which is OK.
But `Vector{Int64}(4)` should be `Vector{Int64}(undef, 4)`, which you've mostly done...probably just accidentally missed this one.

**Comment by cjprybol** [comment]
Thank you for helping to track those down! I've tagged a new 1.0 compatible release that should have addressed all of the 0.7 deprecations https://github.com/cjprybol/uCSV.jl/releases/tag/v0.1.1. The PR for updating METADATA is available here https://github.com/JuliaLang/METADATA.jl/pull/18617
